test/unit/ui.js

```diff
@@ -212,5 +182,143 @@ describe('ui class', () => {
         assert.calledOnce(testCart.destroy);
       });
     });
+
+    describe('cart methods', () => {
+      describe('createCart', () => {
+        let initStub;
+
+        beforeEach(() => {
+          initStub = sinon.stub(Cart.prototype, 'init').resolves();
+        });
+
+        afterEach(() => {
+          initStub.restore();
+        });
+
+        describe('when no cart exists', () => {
+          it('creates a new cart', async () => {
+            await ui.createCart({options: {}});
+            assert.equal(1, ui.components.cart.length);
+            assert.instanceOf(ui.components.cart[0], Cart);
+            assert.calledOnce(initStub);
+            ui.destroyComponent('cart', ui.components.cart[0].model.id);
+          });
+        });
+
+        describe('when a cart exists', () => {
+          let createTogglesStub;
+
+          beforeEach(() => {
+            createTogglesStub = sinon.stub().resolves();
+            ui.components.cart = [{
+              toggles: [1, 2],
+              createToggles: createTogglesStub,
+            }];
+          });
+
+          it('does not create a second cart', async () => {
+            await ui.createCart({options: {}});
+            assert.equal(1, ui.components.cart.length);
+            assert.notCalled(initStub);
+          });
+
+          it('creates toggles for the first cart if the first cart has less toggles than the config', async () => {
+            const createCartConfig = {toggles: [1, 2, 3]};
+            await ui.createCart(createCartConfig);
+            assert.calledOnce(createTogglesStub);
+            assert.calledWith(createTogglesStub, createCartConfig);
+          });
+
+          it('does not create toggles for the first cart if the first cart has an equal amount of toggles to the config', async () => {
+            await ui.createCart({toggles: [1, 2]});
+            assert.notCalled(createTogglesStub);
+          });
+
+          it('does not create toggles for the first cart if the first cart has more toggles than the config', async () => {
+            await ui.createCart({toggles: [1]});
+            assert.notCalled(createTogglesStub);
+          });
+
+          it('returns the first cart', async () => {
+            const createCartConfig = {toggles: [1, 2, 3]};
+            const response = await ui.createCart(createCartConfig);
+            assert.equal(response, ui.components.cart[0]);
+          });
+        });
+      });
+
+      describe('closeCart', () => {
+        it('calls cart.close for every visible cart and calls restoreFocus after each close', () => {
```

Comment:
Let's rewrite the descriptions to remove direct references to the functions called and focus the message on the intended behaviour.